test/test.go

```diff
@@ -201,7 +201,7 @@ func RunJetStreamServerWithPorts(port, monitorPort int, domain string) *server.S
 	opts.JetStream = true
 	opts.JetStreamDomain = domain
 	tdir, _ := os.MkdirTemp(tempRoot, "js-storedir-")
-	opts.StoreDir = filepath.Dir(tdir)
+	opts.StoreDir = tdir
```

Comment:
IMHO, this was a bug. It always selected the parent directory of the `tdir`, effectively re-using the same director always.

Also: I had to create that parent directory by hand. Are you sure that is intended?